helm/values.yaml

```diff
@@ -8,6 +8,9 @@ deployment:
   # imagePullSecrets:
   # - name: "my-container-secret"
   failurePolicy: Fail  # use 'Ignore' to fail open if Connaisseur becomes unavailable
+  # Use 'reinvocationPolicy: IfNeeded' to be called again as part of the admission evaluation if the object
```

Comment:
Not having thought on it for a long time, it may lead to different policies being executed by Connaisseur. I _assume_
```
policy:
  - pattern: "*:*"
    validator: deny
  - pattern: "my-image:my-tag"
    validator: allow
```
would mutate to `my-image@sha` in the first invocation and deny the second. If that's the case, it should be mentioned in the comment for the option that applied policies may change

Reply:
Oh yes, I didn't think about this situation. Do you have any suggestion?

Reply:
Shouldn't Connaisseur "automatically" translate from the tag to the SHA and allow the SHA?

Reply:
It currently cannot and I would think Connaisseur can conceptually not handle this:
As an example, lets say you have a policy for `*:debug` which always denies since you don't want debug images in your cluster and a policy for `*@sha:123` to always allow as part of a allowlisting baseimages.
If presented with the `*@sha123` image, tagged with `debug` and put into a k8s manifest in the tag form, Connaisseur has to honor your decision to deny it, whereas if you had deployed the image@sha it has to honor your decision to allowlist the image.

For your particular problem, iirc you can also sign digests (at least our notary v1 impl has `__search_image_targets_for_digest` and I think cosign does it mainly), so you could for each incoming image sign both tag and digest and make the policy the same for both tags and digests, i.e. `pattern: yourregistryandimage:*`

Reply:
So my PR here is still useful, right? What I need to do is to make sure that both `my.private.registry/image:tag` ang `my.private.registry/image@sha256:hash-of-tag` are signed, right?

Reply:
Yes. My comment wasn't meant to dissuade the PR, I do see the use case. More along the line of, the comment for the reinvocationPolicy should highlight the problem that can arise

Maybe `Note that if Connaisseur is invoked a second time, the policy to be applied might change in between. Make sure, your policies are set up to handle multiple mutations of the image originally specified in the manifest.` or something